File: frontend/src/api/contracts.ts
import { z } from "zod";

import type { components } from "./openapi.generated";

export const platformSchema = z.enum(["facebook", "instagram", "tiktok"]);
export type Platform = z.infer<typeof platformSchema>;

export const authUserSchema = z.object({
  authenticated: z.literal(true),
  user_id: z.string(),
  email: z.string().email().nullable(),
  source_system: z.literal("accumulate").nullable(),
  brand_id: z.string(),
  role: z.string(),
  app_role: z.string().nullable(),
  access_mode: z.enum(["read", "write"]),
  settings_visible: z.boolean(),
  integrations_visible: z.boolean(),
  is_internal_staff: z.boolean(),
  expires_at: z.string(),
  revoked: z.literal(false),
});
export type AuthUser = z.infer<typeof authUserSchema> & components["schemas"]["AuthMeResponse"];

export const brandScopeSchema = z.object({
  requested_brand_id: z.string(),
  rollup: z.boolean(),
  resolved_brand_ids: z.array(z.string()),
});

export const workspaceBrandSchema = z.object({
  brand_id: z.string(),
  name: z.string().nullable(),
  parent_brand_id: z.string().nullable(),
  visibility: z.enum(["active", "hidden_parent"]),
  access_mode: z.enum(["read", "write"]).nullable(),
  role: z.string().nullable(),
});
export type WorkspaceBrand = z.infer<typeof workspaceBrandSchema>;

export const brandWorkspaceSchema = z.object({
  default_brand_id: z.string(),
  brands: z.array(workspaceBrandSchema),
  families: z.array(
    z.object({
      root_brand_id: z.string(),
      brand_ids: z.array(z.string()),
    }),
  ),
  scope: brandScopeSchema,
});
export type BrandWorkspace = z.infer<typeof brandWorkspaceSchema> &
  components["schemas"]["BrandWorkspace"];

export const capabilityStatusSchema = z.enum([
  "unsupported",
  "not_approved",
  "not_configured",
  "blocked_configuration",
  "manual_activation_required",
  "partial",
  "available",
]);
export const capabilitySchema = z.enum(["profile", "content", "comments", "audience"]);

export const workspaceCapabilitiesSchema = z.object({
  scope: brandScopeSchema,
  platforms: z.array(
    z.object({
      platform: platformSchema,
      capabilities: z.array(
        z.object({
          platform: platformSchema,
          capability: capabilitySchema,
          status: capabilityStatusSchema,
          reason: z.string(),
        }),
      ),
      linked_account_count: z.number().int().nonnegative(),
      navigation_available: z.boolean(),
    }),
  ),
  permissions: z.object({
    settings_visible: z.boolean(),
    integrations_visible: z.boolean(),
    internal_audit_visible: z.boolean(),
    rollup_available: z.boolean(),
    operation_mutation_available: z.boolean(),
    tiktok_connection_manage: z.boolean(),
    meta_connection_manage: z.boolean(),
  }),
  runtime: z.object({
    mode: z.enum([
      "development",
      "dormant",
      "staging",
      "standalone_ready",
      "active",
    ]),
    writes_enabled: z.boolean(),
    automated_schedule_available: z.boolean(),
  }),
});
export type WorkspaceCapabilities = z.infer<typeof workspaceCapabilitiesSchema> &
  components["schemas"]["WorkspaceCapabilitiesResponse"];

export const reportingAccountSchema = z.object({
  account_id: z.number().int(),
  brand_id: z.string(),
  platform: platformSchema,
  external_id: z.string(),
  display_name: z.string(),
  status: z.string(),
  connection_state: z.string(),
  health_status: z.string(),
  backfill_status: z.string(),
  link_status: z.string(),
  nightly_enabled: z.boolean(),
  last_synced_at: z.string().nullable(),
});
export type ReportingAccount = z.infer<typeof reportingAccountSchema> &
  components["schemas"]["ReportingAccount"];

export const platformAccountsSchema = z.object({
  meta: brandScopeSchema,
  platform: platformSchema,
  accounts: z.array(reportingAccountSchema),
});

export const metricIdSchema = z.enum([
  "followers",
  "following",
  "new_followers",
  "follows",
  "unfollows",
  "followers_net",
  "reach",
  "reach_paid",
  "reach_organic",
  "views",
  "views_paid",
  "views_organic",
  "interactions",
  "engagement_rate",
  "page_views",
  "profile_views",
  "website_clicks",
  "total_actions",
  "reactions",
  "media_count",
  "video_views_total",
  "video_views_change",
  "video_likes_total",
  "video_comments_total",
  "video_shares_total",
  "video_engagements_total",
  "video_engagement_rate",
]);
export type MetricId = z.infer<typeof metricIdSchema>;

export const dataStatusSchema = z.enum(["available", "partial", "unavailable"]);
export type DataStatus = z.infer<typeof dataStatusSchema>;

export const availabilityStatusSchema = z.enum([
  "available",
  "partial",
  "pending",
  "provider_unavailable",
  "unavailable",
]);

const dashboardMetaSchema = z.object({
  dashboard_id: z.string(),
  platform: platformSchema.nullable(),
  requested_brand_id: z.string(),
  rollup: z.boolean(),
  resolved_brand_ids: z.array(z.string()),
  resolved_account_ids: z.array(z.number().int()),
  date_range: z.object({
    start_on: z.string(),
    end_on: z.string(),
    key: z.string(),
  }),
  generated_at: z.string(),
  last_sync_at: z.string().nullable(),
  freshness: z.enum(["fresh", "stale", "outdated", "never_synced"]),
  observed_days: z.number().int().nonnegative(),
  expected_days: z.number().int().positive(),
  data_status: dataStatusSchema,
  warnings: z.array(z.string()),
});

export const dashboardMetricSchema = z.object({
  metric_id: metricIdSchema,
  value: z.number().nullable(),
  previous_value: z.number().nullable(),
  delta_pct: z.number().nullable(),
  semantic_type: z.enum(["snapshot", "flow", "cumulative", "ratio"]),
  unit: z.enum(["count", "ratio"]),
  data_status: dataStatusSchema,
  methodology: z.string().min(1),
  availability_reason: z.string().nullable(),
});
export type DashboardMetric = z.infer<typeof dashboardMetricSchema>;

const dashboardSeriesSchema = z.object({
  metric_id: metricIdSchema,
  semantic_type: z.enum(["snapshot", "flow", "cumulative", "ratio"]),
  points: z.array(z.object({ observed_on: z.string(), value: z.number() })),
  methodology: z.string().min(1),
});
export type DashboardSeries = z.infer<typeof dashboardSeriesSchema>;

const dashboardBreakdownSchema = z.object({
  metric_id: metricIdSchema,
  dimension: z.string(),
  items: z.array(
    z.object({
      key: z.string(),
      value: z.number(),
      percentage: z.number().nullable(),
    }),
  ),
});
export type DashboardBreakdown = z.infer<typeof dashboardBreakdownSchema>;

export const dashboardContentSchema = z.object({
  account_id: z.number().int(),
  external_content_id: z.string(),
  content_type: z.string(),
  permalink: z.string(),
  message: z.string(),
  media_url: z.string(),
  published_at: z.string().nullable(),
  likes_count: z.number().int(),
  comments_count: z.number().int(),
  shares_count: z.number().int(),
  interactions: z.number().int(),
  views: z.number().nullable(),
  reach: z.number().nullable(),
  cover_url: z.string().nullable(),
  thumbnail_url: z.string().nullable(),
  cover_candidates: z.array(z.string()),
  thumbnail_candidates: z.array(z.string()),
  media_url_candidates: z.array(z.string()),
  full_video_watched_rate: z.number().nullable(),
  total_time_watched: z.number().nullable(),
  average_time_watched: z.number().nullable(),
  data_status: dataStatusSchema,
});
export type DashboardContent = z.infer<typeof dashboardContentSchema>;

const communitySchema = z.object({
  total_comments: z.number().int().nonnegative(),
  answered_comments: z.number().int().nonnegative(),
  unanswered_comments: z.number().int().nonnegative(),
  comment_likes: z.number().int().nonnegative(),
  data_status: dataStatusSchema,
  top_commenters: z.array(z.object({
    name: z.string(),
    comments: z.number().int().nonnegative(),
    likes: z.number().int().nonnegative(),
  })),
  top_liked_comments: z.array(z.object({
    name: z.string(),
    comment: z.string(),
    likes: z.number().int().nonnegative(),
    replies: z.number().int().nonnegative(),
  })),
});

const dashboardNamedValueSchema = z.object({
  name: z.string(),
  value: z.number(),
});

export const dashboardContentSummarySchema = z.object({
  total: z.number().int().nonnegative(),
  by_type: z.array(dashboardNamedValueSchema),
  reach_by_type: z.array(dashboardNamedValueSchema),
  views_by_type: z.array(dashboardNamedValueSchema),
  data_status: dataStatusSchema,
});

const dashboardSourceValuesSchema = z.object({
  organic: z.number().nullable(),
  paid: z.number().nullable(),
  data_status: dataStatusSchema,
});

export const dashboardSourceBreakdownSchema = z.object({
  organic_only: z.boolean(),
  paid_available: z.boolean(),
  views: dashboardSourceValuesSchema.nullable(),
  reach: dashboardSourceValuesSchema.nullable(),
  data_status: dataStatusSchema,
}).nullable();

export const dashboardMetricMethodologySchema = z.object({
  follower_flow: z.string().min(1),
  engagement_rate: z.string().min(1),
  reach: z.string().min(1),
});

export const dashboardAudienceCapabilitiesSchema = z.object({
  source: z.string().nullable(),
  geo: availabilityStatusSchema,
  age_gender: availabilityStatusSchema,
  activity: availabilityStatusSchema,
});

const dashboardStorySummarySchema = z.object({
  count: z.number().int().nonnegative(),
  views: z.number().nullable(),
  reach: z.number().nullable(),
  interactions: z.number().nullable(),
  replies: z.number().nullable(),
  completion_rate: z.number().nullable(),
  data_status: dataStatusSchema,
});

const dashboardStoryTrendSchema = z.object({
  labels: z.array(z.string()),
  views: z.array(z.number().nullable()),
  reach: z.array(z.number().nullable()),
  data_status: dataStatusSchema,
});

const dashboardStoryNavigationSchema = z.object({
  taps_forward: z.number().nullable(),
  taps_back: z.number().nullable(),
  swipe_forward: z.number().nullable(),
  exits: z.number().nullable(),
  data_status: dataStatusSchema,
});

const dashboardStoryActionsSchema = z.object({
  replies: z.number().nullable(),
  shares: z.number().nullable(),
  profile_visits: z.number().nullable(),
  follows: z.number().nullable(),
  sticker_taps: z.number().nullable(),
  saves: z.number().nullable(),
  data_status: dataStatusSchema,
});

export const dashboardStoryItemSchema = z.object({
  content_id: z.string(),
  title: z.string(),
  cover_url: z.string(),
  permalink: z.string(),
  created_time: z.string().nullable(),
  views: z.number().nullable(),
  reach: z.number().nullable(),
  interactions: z.number().nullable(),
  replies: z.number().nullable(),
  shares: z.number().nullable(),
  profile_visits: z.number().nullable(),
  follows: z.number().nullable(),
  sticker_taps: z.number().nullable(),
  saves: z.number().nullable(),
  taps_forward: z.number().nullable(),
  taps_back: z.number().nullable(),
  swipe_forward: z.number().nullable(),
  exits: z.number().nullable(),
  navigation: z.number().nullable(),
  completion_rate: z.number().nullable(),
  data_status: dataStatusSchema,
});
export type DashboardStoryItem = z.infer<typeof dashboardStoryItemSchema>;

export const dashboardStoriesSchema = z.object({
  summary: dashboardStorySummarySchema,
  previous_summary: dashboardStorySummarySchema,
  trend: dashboardStoryTrendSchema,
  navigation: dashboardStoryNavigationSchema,
  actions: dashboardStoryActionsSchema,
  items: z.array(dashboardStoryItemSchema),
  data_status: dataStatusSchema,
}).nullable();
export type DashboardStories = Exclude<z.infer<typeof dashboardStoriesSchema>, null>;

export const platformDashboardSchema = z.object({
  meta: dashboardMetaSchema,
  metrics: z.array(dashboardMetricSchema),
  series: z.array(dashboardSeriesSchema),
  breakdowns: z.array(dashboardBreakdownSchema),
  content: z.array(dashboardContentSchema),
  community: communitySchema,
  top_hashtags: z.array(z.object({
    name: z.string(),
    count: z.number().int().nonnegative(),
  })),
  content_summary: dashboardContentSummarySchema,
  source_breakdown: dashboardSourceBreakdownSchema,
  metric_methodology: dashboardMetricMethodologySchema,
  audience_capabilities: dashboardAudienceCapabilitiesSchema,
  stories: dashboardStoriesSchema,
});
export type PlatformDashboard = z.infer<typeof platformDashboardSchema> &
  components["schemas"]["PlatformDashboard"];

export const overviewDashboardSchema = z.object({
  meta: dashboardMetaSchema,
  metrics: z.array(dashboardMetricSchema),
  platforms: z.array(platformDashboardSchema),
  content: z.array(dashboardContentSchema),
  community: communitySchema,
});
export type OverviewDashboard = z.infer<typeof overviewDashboardSchema> &
  components["schemas"]["OverviewDashboard"];

const settingsBrandSchema = z.object({
  brand_id: z.string(),
  name: z.string().nullable(),
  parent_brand_id: z.string().nullable(),
  visibility: z.string(),
  access_mode: z.string().nullable(),
  role: z.string().nullable(),
  linked_account_count: z.number().int().nonnegative(),
  last_sync_at: z.string().nullable(),
});
export type SettingsBrand = z.infer<typeof settingsBrandSchema>;

export const settingsBrandsSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(settingsBrandSchema),
});
export type SettingsBrands = z.infer<typeof settingsBrandsSchema>;

export const socialAccountsSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(reportingAccountSchema),
});
export type SocialAccounts = z.infer<typeof socialAccountsSchema>;

const brandLinkSchema = z.object({
  brand_id: z.string(),
  platform: platformSchema,
  account_id: z.number().int(),
  external_id: z.string(),
  display_name: z.string(),
  link_status: z.string(),
});
export type BrandLink = z.infer<typeof brandLinkSchema>;

export const brandLinksSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(brandLinkSchema),
});
export type BrandLinks = z.infer<typeof brandLinksSchema>;

const connectionSchema = z.object({
  connection_id: z.number().int(),
  brand_id: z.string(),
  platform: platformSchema,
  state: z.string(),
  expires_at: z.string().nullable(),
  projected_at: z.string().nullable(),
});
export type ReportingConnection = z.infer<typeof connectionSchema>;

export const connectionsSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(connectionSchema),
});
export type ReportingConnections = z.infer<typeof connectionsSchema>;

const syncJobSchema = z.object({
  job_id: z.number().int(),
  brand_id: z.string(),
  account_id: z.number().int().nullable(),
  platform: platformSchema,
  stage: z.string(),
  status: z.string(),
  scheduled_for: z.string(),
  started_at: z.string().nullable(),
  finished_at: z.string().nullable(),
  error_code: z.string().nullable(),
});
export type ReportingSyncJob = z.infer<typeof syncJobSchema>;

export const syncJobsSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(syncJobSchema),
});
export type ReportingSyncJobs = z.infer<typeof syncJobsSchema>;

export const insightSchema = z.object({
  insight_id: z.number().int(),
  brand_id: z.string(),
  status: z.string(),
  date_from: z.string().nullable(),
  date_to: z.string().nullable(),
  summary: z.string().nullable(),
  recommendations: z.string().nullable(),
  connector_analysis: z.string().nullable(),
  anomalies: z.string().nullable(),
  platform_evaluations: z.string().nullable(),
  model: z.string().nullable(),
  error_message: z.string().nullable(),
  created_by_user_sub: z.string().nullable(),
  created_at: z.string(),
  completed_at: z.string().nullable(),
});
export type ReportingInsight = z.infer<typeof insightSchema>;

export const insightsSchema = z.object({
  meta: brandScopeSchema,
  items: z.array(insightSchema),
});

export const aiSummaryLimitSchema = z.object({
  provider_configured: z.boolean(),
  can_generate: z.boolean(),
  reason: z.string(),
  weekly_limit: z.number().int().positive(),
  used: z.number().int().nonnegative(),
  remaining: z.number().int().nonnegative(),
  window_days: z.number().int().positive(),
  last_generated_at: z.string().nullable(),
  next_available_at: z.string().nullable(),
  generation_in_progress: z.boolean(),
});
export type AiSummaryLimit = z.infer<typeof aiSummaryLimitSchema>;

export const readinessSchema = z.object({
  status: z.string(),
  runtime_mode: z.string(),
  writes_enabled: z.boolean(),
  database_configured: z.boolean(),
  scope: brandScopeSchema.nullable().optional(),
  platforms: z
    .array(
      z.object({
        platform: platformSchema,
        account_count: z.number().int().nonnegative(),
        last_sync_at: z.string().nullable(),
        pending_job_count: z.number().int().nonnegative(),
      }),
    )
    .default([]),
});
export type OperationsReadiness = z.infer<typeof readinessSchema>;

export const auditSchema = z.object({
  meta: brandScopeSchema,
  status: z.string(),
  reason: z.string(),
  items: z.array(z.unknown()),
});
export type ReportingAudit = z.infer<typeof auditSchema>;

export const tiktokConnectionSchema = z.object({
  meta: brandScopeSchema,
  state: z.string(),
  connection: connectionSchema.nullable(),
  capabilities: z.array(
    z.object({
      platform: platformSchema,
      capability: capabilitySchema,
      status: capabilityStatusSchema,
      reason: z.string(),
    }),
  ),
  checked_at: z.string(),
});
export type TikTokConnection = z.infer<typeof tiktokConnectionSchema>;

export const tiktokActivationReadinessSchema = z.object({
  handoff_ready: z.literal(true),
  brand_id: z.string(),
  launch_target: z.literal("tiktok_owner_activation"),
  fresh_until: z.string(),
  runtime_mode: z.string(),
  writes_enabled: z.boolean(),
  connection_state: z.string(),
  oauth_start_available: z.boolean(),
  reason: z.string(),
  checked_at: z.string(),
});
export type TikTokActivationReadiness = z.infer<typeof tiktokActivationReadinessSchema> &
  components["schemas"]["TikTokActivationReadinessResponse"];

export const tiktokSelfServiceReadinessSchema = z.object({
  brand_id: z.string(),
  can_manage: z.boolean(),
  connection_state: z.string(),
  linked_account_count: z.number().int().nonnegative(),
  oauth_start_available: z.boolean(),
  reason: z.string(),
  runtime_mode: z.string(),
  writes_enabled: z.boolean(),
  checked_at: z.string(),
});
export type TikTokSelfServiceReadiness = z.infer<typeof tiktokSelfServiceReadinessSchema>;

export const tiktokSelfServiceStartSchema = z.object({
  authorization_url: z.string().url(),
  expires_at: z.string(),
});
export type TikTokSelfServiceStart = z.infer<typeof tiktokSelfServiceStartSchema>;

export const metaDiscoverySchema = z.object({
  connection_id: z.number().int().positive(),
  platform: z.enum(["facebook", "instagram"]),
  external_id: z.string(),
  display_name: z.string(),
  status: z.string(),
});
export type MetaDiscovery = z.infer<typeof metaDiscoverySchema>;

export const metaLinkedAccountSchema = z.object({
  platform: z.enum(["facebook", "instagram"]),
  external_id: z.string(),
  display_name: z.string(),
});

export const metaSelfServiceReadinessSchema = z.object({
  brand_id: z.string(),
  can_manage: z.boolean(),
  connection_state: z.string(),
  facebook_linked_count: z.number().int().nonnegative(),
  instagram_linked_count: z.number().int().nonnegative(),
  linked_accounts: z.array(metaLinkedAccountSchema),
  discoveries: z.array(metaDiscoverySchema),
  oauth_start_available: z.boolean(),
  reason: z.string(),
  runtime_mode: z.string(),
  writes_enabled: z.boolean(),
  checked_at: z.string(),
});
export type MetaSelfServiceReadiness = z.infer<typeof metaSelfServiceReadinessSchema>;

export const metaSelfServiceStartSchema = z.object({
  authorization_url: z.string().url(),
  expires_at: z.string(),
});

export const metaLinkResponseSchema = z.object({
  connection_id: z.number().int().positive(),
  linked_count: z.number().int().positive(),
  connection_state: z.literal("connected"),
});

export const reportJobSchema = z.object({
  job_id: z.string().min(1),
  state: z.enum(["queued", "running", "ready", "failed"]),
  progress: z.number().int().min(0).max(100),
  stage: z.string(),
  filename: z.string().nullable(),
  created_at: z.string(),
  expires_at: z.string().nullable(),
  error_code: z.string().nullable(),
});
export type ReportJob = z.infer<typeof reportJobSchema> &
  components["schemas"]["ReportJobResponse"];

export const apiProblemSchema = z.object({
  detail: z.unknown().optional(),
});

File: backend/app/api/contracts.py
"""Typed response contracts shared by the Phase 6 API surface."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from typing import Literal

from app.application.ports.ai_summary import AiSummaryLimitStatus
from app.application.ports.reporting import (
    ReportingAccount,
    ReportingConnection,
    ReportingInsight,
    ReportingSyncJob,
)
from app.capabilities import CapabilityRecord
from app.core import RuntimeMode
from app.domain.authority import BrandScope
from app.domain.platforms import PlatformId


@dataclass(frozen=True)
class AuthMeResponse:
    authenticated: bool
    user_id: str
    email: str | None
    source_system: str | None
    brand_id: str
    role: str
    app_role: str | None
    access_mode: str
    settings_visible: bool
    integrations_visible: bool
    is_internal_staff: bool
    expires_at: datetime
    revoked: bool


@dataclass(frozen=True)
class PlatformAccountsResponse:
    meta: BrandScope
    platform: PlatformId
    accounts: tuple[ReportingAccount, ...]


@dataclass(frozen=True)
class SettingsBrandItem:
    brand_id: str
    name: str | None
    parent_brand_id: str | None
    visibility: str
    access_mode: str | None
    role: str | None
    linked_account_count: int
    last_sync_at: datetime | None


@dataclass(frozen=True)
class SettingsBrandsResponse:
    meta: BrandScope
    items: tuple[SettingsBrandItem, ...]


@dataclass(frozen=True)
class SocialAccountsResponse:
    meta: BrandScope
    items: tuple[ReportingAccount, ...]


@dataclass(frozen=True)
class BrandLinkItem:
    brand_id: str
    platform: PlatformId
    account_id: int
    external_id: str
    display_name: str
    link_status: str


@dataclass(frozen=True)
class BrandLinksResponse:
    meta: BrandScope
    items: tuple[BrandLinkItem, ...]


@dataclass(frozen=True)
class ConnectionsResponse:
    meta: BrandScope
    items: tuple[ReportingConnection, ...]


@dataclass(frozen=True)
class SyncJobsResponse:
    meta: BrandScope
    items: tuple[ReportingSyncJob, ...]


@dataclass(frozen=True)
class AuditResponse:
    meta: BrandScope
    status: str
    reason: str
    items: tuple[object, ...]


@dataclass(frozen=True)
class TikTokConnectionResponse:
    meta: BrandScope
    state: str
    connection: ReportingConnection | None
    capabilities: tuple[CapabilityRecord, ...]
    checked_at: datetime


@dataclass(frozen=True)
class TikTokActivationReadinessResponse:
    handoff_ready: bool
    brand_id: str
    launch_target: str
    fresh_until: datetime
    runtime_mode: RuntimeMode
    writes_enabled: bool
    connection_state: str
    oauth_start_available: bool
    reason: str
    checked_at: datetime


@dataclass(frozen=True)
class TikTokSelfServiceReadinessResponse:
    brand_id: str
    can_manage: bool
    connection_state: str
    linked_account_count: int
    oauth_start_available: bool
    reason: str
    runtime_mode: RuntimeMode
    writes_enabled: bool
    checked_at: datetime


@dataclass(frozen=True)
class TikTokSelfServiceStartResponse:
    authorization_url: str
    expires_at: datetime


@dataclass(frozen=True)
class MetaDiscoveryItem:
    connection_id: int
    platform: PlatformId
    external_id: str
    display_name: str
    status: str


@dataclass(frozen=True)
class MetaLinkedAccountItem:
    platform: PlatformId
    external_id: str
    display_name: str


@dataclass(frozen=True)
class MetaSelfServiceReadinessResponse:
    brand_id: str
    can_manage: bool
    connection_state: str
    facebook_linked_count: int
    instagram_linked_count: int
    linked_accounts: tuple[MetaLinkedAccountItem, ...]
    discoveries: tuple[MetaDiscoveryItem, ...]
    oauth_start_available: bool
    reason: str
    runtime_mode: RuntimeMode
    writes_enabled: bool
    checked_at: datetime


@dataclass(frozen=True)
class MetaSelfServiceStartResponse:
    authorization_url: str
    expires_at: datetime


@dataclass(frozen=True)
class MetaLinkResponse:
    connection_id: int
    linked_count: int
    connection_state: str


@dataclass(frozen=True)
class MetaRefreshResponse:
    connection_id: int
    facebook_count: int
    instagram_count: int
    discovered_count: int


@dataclass(frozen=True)
class InsightsResponse:
    meta: BrandScope
    items: tuple[ReportingInsight, ...]


@dataclass(frozen=True)
class AiSummaryLimitResponse:
    provider_configured: bool
    can_generate: bool
    reason: str
    weekly_limit: int
    used: int
    remaining: int
    window_days: int
    last_generated_at: datetime | None
    next_available_at: datetime | None
    generation_in_progress: bool

    @classmethod
    def from_status(cls, status: AiSummaryLimitStatus) -> AiSummaryLimitResponse:
        return cls(**status.__dict__)


@dataclass(frozen=True)
class CapabilityPlatform:
    platform: PlatformId
    capabilities: tuple[CapabilityRecord, ...]
    linked_account_count: int
    navigation_available: bool


@dataclass(frozen=True)
class WorkspacePermissions:
    settings_visible: bool
    integrations_visible: bool
    internal_audit_visible: bool
    rollup_available: bool
    operation_mutation_available: bool
    tiktok_connection_manage: bool
    meta_connection_manage: bool


@dataclass(frozen=True)
class RuntimeCapabilities:
    mode: RuntimeMode
    writes_enabled: bool
    automated_schedule_available: bool


@dataclass(frozen=True)
class ReadinessPlatform:
    platform: PlatformId
    account_count: int
    last_sync_at: datetime | None
    pending_job_count: int


@dataclass(frozen=True)
class OperationsReadinessResponse:
    status: str
    runtime_mode: RuntimeMode
    writes_enabled: bool
    database_configured: bool
    scope: BrandScope | None = None
    platforms: tuple[ReadinessPlatform, ...] = ()


@dataclass(frozen=True)
class ReportJobResponse:
    job_id: str
    state: Literal["queued", "running", "ready", "failed"]
    progress: int
    stage: str
    filename: str | None
    created_at: datetime
    expires_at: datetime | None
    error_code: str | None


@dataclass(frozen=True)
class WorkspaceCapabilitiesResponse:
    scope: BrandScope
    platforms: tuple[CapabilityPlatform, ...]
    permissions: WorkspacePermissions
    runtime: RuntimeCapabilities


__all__ = [
    "AuditResponse",
    "AuthMeResponse",
    "BrandLinkItem",
    "BrandLinksResponse",
    "CapabilityPlatform",
    "ConnectionsResponse",
    "InsightsResponse",
    "MetaDiscoveryItem",
    "MetaLinkedAccountItem",
    "MetaLinkResponse",
    "MetaRefreshResponse",
    "MetaSelfServiceReadinessResponse",
    "MetaSelfServiceStartResponse",
    "OperationsReadinessResponse",
    "PlatformAccountsResponse",
    "RuntimeCapabilities",
    "ReadinessPlatform",
    "ReportJobResponse",
    "SettingsBrandItem",
    "SettingsBrandsResponse",
    "SocialAccountsResponse",
    "SyncJobsResponse",
    "TikTokConnectionResponse",
    "TikTokActivationReadinessResponse",
    "TikTokSelfServiceReadinessResponse",
    "TikTokSelfServiceStartResponse",
    "WorkspaceCapabilitiesResponse",
    "WorkspacePermissions",
]
